c++/src/kj/compat/http.h

```diff
@@ -1089,12 +1091,14 @@ inline kj::StringPtr HttpHeaderTable::idToString(HttpHeaderId id) const {
 
 inline kj::Maybe<kj::StringPtr> HttpHeaders::get(HttpHeaderId id) const {
   id.requireFrom(*table);
+  assertKnownIndexedHeaderId(id);
```

Comment:
This method could previously be optimized down to an array index, but now it involves an out-of-line function call. That seems kind of sad. Can we make this a debug-only check? (Or assert in the HttpHeaders constructor instead, as discussed.)

Reply:
Yeah, I agree that it's a bit sad. I was conflicted as well. Out of curiosity, do you happen to remember why these functions were in the header? Maybe similar hopes for inline optimization?

Anyway, since you're fine with the ctor assertion, I'll do that instead.

Reply:
`get()` is intended to be inline-able, yes. A major design goal here -- the reason we have the whole header ID registration system -- was that accessing headers should be extremely cheap, compared to typical HTTP libraries that do a name hash lookup for every access.